src/ApplicationStatus.hpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2015, 2016, 2017, 2018, 2019, 2020, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of Intel Corporation nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef APPLICATIONSTATUS_HPP_INCLUDE
+#define APPLICATIONSTATUS_HPP_INCLUDE
+
+#include <cstdint>
+
+#include <memory>
+#include <vector>
+#include <set>
+
+#include "Helper.hpp"
+
+namespace geopm
+{
+    class SharedMemory;
+    class PlatformTopo;
+
+    /// @brief Object that encapsulates application process
+    ///        information such as the process ID or region hint.
+    ///        There will be one ApplicationStatus for the node (board
+    ///        domain) on each side of the shared memory.
+    class ApplicationStatus
+    {
+        public:
+            virtual ~ApplicationStatus() = default;
+
+            /// @brief Set the current hint bits for a CPU.
+            virtual void set_hint(int cpu_idx, uint64_t hints) = 0;
+            /// @brief Get the current hint bits for a CPU.
+            virtual uint64_t get_hint(int cpu_idx) const = 0;
+            /// @brief Get the current hint bits for every CPU.
+            virtual std::vector<uint64_t> get_hint(void) const = 0;
+            /// @brief Reset the total work units for a single CPU to
```

Comment:
Providing doxygen for @brief, summary, @param and @return would be nice, but maybe not critical for this PR.